### ISSUE TITLE
Icon improvement

### ISSUE THREAD

**Issue by kiranetph** [Serhioromano/cobalt-9#58]
The default icons in Cobalt are not really "up-to-date" anymore.
I would suggest to improve the icon select. My suggestion would be a 3 options to select icons/graphics in cobalt-- whatever is filled out will be used.
1) The existing way with the icon collection to keep compatibility with existing installations.
2) An optional text field for font icons with additional icon selector (see gantry 5 screenshot below)
3) an image select control, to use uploaded images

to 2)
![2016-01-27_14-50-38](https://cloud.githubusercontent.com/assets/15226809/12606238/ba9105ce-c506-11e5-99aa-10e5522c166a.png)

Can be a conditional parameter for the icon field with options "None", "Icon", "Font Icon", "Image" and the the individual selector.

**Comment by drinkmaker** [comment]
Like the existing system of selecting icons. It is simple. And it's great. 
Indeed is perhaps should update icons. You can even add the ability to upload your images.

**Comment by kiranetph** [comment]
@drinkmaker 
Yes, keeping the existing icons (just they might not fit into modern templates so much anymore), but giving additional options for own uploaded icons/images (different sizes) and also font icons (like in the example above).
There can be a parameter like:
![2016-01-27_16-38-46](https://cloud.githubusercontent.com/assets/15226809/12608085/9f0d4caa-c514-11e5-838e-3bca877c0219.png)
Then depending on the selection (conditional parameter #13 ), the respective dialog will appear.

**Comment by drinkmaker** [comment]
@kiranetph 
I understood you. It's complicated:
1. What font to use? Which version?
2. It can conflict with icon-font in template
3. how to deal with the Custom icon-fonts?
We can not solve all possible situations. Let's not chase it. Basic icons you can choose from a pictures. For everything else there's ~~Mastercard~~ custom templates :)

**Comment by kiranetph** [comment]
@drinkmaker 
to 1) therefore there is a textbox entry for font ....
most templates have font-awesome included. if it is font awesome 4.x just use "fa fa-trash fa-lg", if it is font awesome then type in "icon-trash icon-large" 
then use it like `<i class="icon-trash icon-large"></i> Delete</a>`
An icon-selector like in gantry5 would be the icing on the cake. 
2) it should not add an additional icon-font, just use what is in the template and put it in the textbox.
3) custom icon fonts would be added to the template by an additional check box parameter "Include custom font" or it is a general template addition (custom template addition) then you just ad the class in the textbox

**Comment by pepperstreet** [comment]
+1000  (enough comments written on the forum ;) )

---

BTW, here is my old topic from June 2013 with some related links: 
http://www.mintjoomla.com/support/community-forum/user-item/81-pepperstreet/48-cobalt-8/2399-category-image-parameter-for-section-icons-webfont.html

**Comment by Serhioromano** [comment]
This is good and modern idea but questions remain as @drinkmaker had asked. What fonts? What icons? What style to place icons.

All these questions about Cobalt look drives me nuts because there is no consistency in Joomla templates. Every provider creates own styles, it is so hard to keep up to look good on every template.

Still, my main point here is the tool for users to customize templates. In cobalt 9 user will be able to edit all template files right in the template manager. I even think of using Monaco editor for that for great on the spot editing experience.
